docs/developers/add_custom_call.md

```diff
@@ -0,0 +1,80 @@
+# Development Guide: Add a Custom Call Operator
+
+In some scenarios, the BladeDISC code generator can not meet user needs,
+e.g. vender library or calling a customized CUDA kernel code. BladeDISC
+provides an interface to call these customized functions, which is called
+**Custom Call Operator**, you can find more design details in
+[Runtime Abstraction Layer Introduction](/docs/developers/runtime_abstraction_layer.md).
+
+## Add a Custom Call Operator Step by Step
+
+To make it easier for developers to understand, this section uses `TransposeOp`
+as an example to introduce how to add a custom call operator step by step.
+
+### Step1: Register Operator in RAL
+
+BladeDISC provides a macro `TAO_RAL_API` to register a custom call operator. To
+make the code structure clearly, please create a new `transpose_impl.cc` file
+under the directory: `tao_compiler/mlir/xla/ral/context/`:
+
+``` c++
+void ral_gpu_transpose_2d(ExecutionContext* ctx, void* stream_handle,
+                          MemRefType<T, 2> input, int dim0, int dim1,
```

Comment:
better to use a local cpu buffer (e.g. memref.alloca) to represent the transpose permutation to support the general case. The metadata fields of conv op is an example.

Reply:
Agree with this comment, and I will update this doc with https://github.com/alibaba/BladeDISC/pull/685 in the near future.